tests/test_upstream_rbus.c

```diff
@@ -0,0 +1,164 @@
+/**
+ *  Copyright 2010-2016 Comcast Cable Communications Management, LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#include <stdio.h>
+#include <rbus.h>
+#include <CUnit/Basic.h>
+#include "../src/upstream.h"
+#include "../src/ParodusInternal.h"
+
+#define CLOUD_CONN_ONLINE "cloud_conn_online_event"
+
+rbusHandle_t handle;
+
+int validate_partner_id(wrp_msg_t *msg, partners_t **partnerIds)
+{
+	return;
+}
+
+int sendUpstreamMsgToServer(void **resp_bytes, size_t resp_size)
+{
+	return;
+}
+//Test case for rbusRegCloudConnectionOnlineEventRbushandle failure 
+void test_regConnOnlineEventRbushandle_failure()
+{
+	int result = regConnOnlineEvent();
+	CU_ASSERT_EQUAL(result, -1);
+}
+
+//Test case for rbusRegCloudConnectionOnlineEvent success
+void test_regConnOnlineEvent_success()
+{
+	subscribeRBUSevent();
+	int result = regConnOnlineEvent();
+	CU_ASSERT_EQUAL(result, 0);
+}
+
+//Test case for rbusRegCloudConnectionOnlineEvent failure
+void test_regConnOnlineEvent_failure()
+{
+	int result = regConnOnlineEvent();
+	result = regConnOnlineEvent();
```

Comment:
why we are doing register twice here? Isn't it same as previous test case?

Reply:
Registered event two time to create error scenario
added comment in code